src/main/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTask.java

```diff
@@ -85,24 +86,44 @@ void stop() {
   /** @see MongoSinkTask#put(Collection) */
   void put(final Collection<SinkRecord> records) {
     if (lastTaskInvocation != null) {
-      statistics.timeSpentOutsidePutTask(lastTaskInvocation);
+      statistics
+          .getBetweenTaskInvocations()
+          .sample(lastTaskInvocation.getElapsedTime(TimeUnit.MILLISECONDS));
     }
-    Timer taskTime = statistics.putTaskInvoked();
-    statistics.recordsReceived(records.size());
+    Timer taskTime = Timer.start();
+    statistics.getRecordsReceived().sample(records.size());
+    trackLatestRecordTimestampOffset(records);
     if (records.isEmpty()) {
       LOGGER.debug("No sink records to process for current poll operation");
     } else {
-      Timer processingTime = statistics.startTimer();
+      Timer processingTime = Timer.start();
       List<List<MongoProcessedSinkRecordData>> batches =
           MongoSinkRecordProcessor.orderedGroupByTopicAndNamespace(
               records, sinkConfig, errorReporter);
-      statistics.putTaskRecordProcessingTime(processingTime);
+      statistics
+          .getRecordsProcessing()
+          .sample(processingTime.getElapsedTime(TimeUnit.MILLISECONDS));
       for (List<MongoProcessedSinkRecordData> batch : batches) {
         bulkWriteBatch(batch);
       }
     }
-    statistics.putTaskTime(taskTime);
-    lastTaskInvocation = statistics.startTimer();
+    statistics.getTaskInvocations().sample(taskTime.getElapsedTime(TimeUnit.MILLISECONDS));
+    lastTaskInvocation = Timer.start();
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(statistics.getName() + ": " + statistics.toJSON());
```

Comment:
I wrote it this way because `toJSON` itself seemed relatively expensive, compared to the concatenation we'd be trying to save on here.

In response to your question about performance, I think it's negligible. We never increment counters in a tight loop. It's effectively an `AtomicLong` increment per stat. Usually this is once per poll, though iirc because of batching, retries, and the like, it might go up to "a few times" per poll.

Reply:
Added a small comment.